### PR TITLE
Api refactor

### PR DESCRIPTION
**Pull request by chris-french**
The big changes are:
- [x] Removing all the Python GIS logic with an [SQL query](https://github.com/codeforkansascity/meep-backend/blob/23797f67d0c469f7b3989f709f49ca3897fd548a/api/app/services/location_marker.py#L47)
- [x] Adding a [spec for the Location/Project APIs](https://github.com/codeforkansascity/meep-backend/blob/api-refactor/docs/spec.md)
- [x] Removing `forms` endpoints
- [x] Refactor Google Maps API geocoding service
- [x] Testing required
  - [x] Need to test new geocoding service
  - [x] Need to update previous tests

Added a script to run pytests:

```
scripts/local/run_tests.sh
```

### CODE REVIEW

**Comment by chris-french** [comment]
@chrisschaaf  @krisstee  I talked to Jordan, and they aren't planning to have users add/updated/delete project data, so I'm going to remove all those `forms` endpoints.

**Comment by chris-french** [comment]
@chrisschaaf @krisstee Ok, I think I'm done with the refactor.

The next to do items are:
- [ ] Update the docker files
- [ ] Update the data pipeline to load MEC data in from a CSV file

I'll leave both of these items to other PRs, so I'm ready to merge if this looks good.

Another to do item is figuring out how to deploy the front-end and back-end. Maybe we can add this repo to the `meep` repo, and create one Docker file and run the full stack (React + Flask + Postgres) on a single machine?

File: scripts/local/run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/bash
```
Comment (chrisschaaf) [inline scripts/local/run_tests.sh]: This seems to cause macos/zsh to complain - `#!/usr/bin/env bash` should cover a wider range of shell setups.

**Comment by chrisschaaf** [comment]
Should the `api/instance` directory have been moved over as well? Or was this what was supposed to be targeted in `.gitignore`? Hitting the following when starting the test server:

```
FileNotFoundError: [Errno 2] Unable to load configuration file (No such file or directory): './src/instance/env.test.cfg'
```

If I replace this file at `src/instance/env.test.cfg` it starts up alright, and all tests pass.

**Comment by chris-french** [comment]
> Should the `api/instance` directory have been moved over as well? Or was this what was supposed to be targeted in `.gitignore`? Hitting the following when starting the test server:
> 
> ```
> FileNotFoundError: [Errno 2] Unable to load configuration file (No such file or directory): './src/instance/env.test.cfg'
> ```
> 
> If I replace this file at `src/instance/env.test.cfg` it starts up alright, and all tests pass.

@chrisschaaf ok, should be fixed. I had added a google api key to the config file and was careful not to commit the key -- should have moved the file and committed the change of directory first before adding the additional api key.

**Pull request merged**